blocks/product-details/product-details.js

```diff
@@ -1,26 +1,254 @@
+    'Magento-Customer-Group': await getConfigValue('commerce-customer-group'),
+    'x-api-key': await getConfigValue('commerce-x-api-key'),
+  });
+
+  // pre-fetch PDP data
+  const sku = getSkuFromUrl();
+  const optionsUIDs = getOptionsUIDsFromUrl();
+
+  const [product, labels] = await Promise.all([
+    PDP.fetchPDPData(sku, { optionsUIDs }),
+    fetchPlaceholders(),
+  ]);
+
+  // Initialize
+  initializers.register(PDP.initialize, {
+    sku,
+    optionsUIDs,
+    langDefinitions: getLangDefinitions(labels),
+    models: {
+      ProductDetails: {
+        initialData: product,
+      },
+    },
+    acdl: true,
+    persistURLParams: true,
+  });
+
+  if (!product) {
+    await loadErrorPage();
+    return Promise.reject();
+  }
+
+  // Layout
+  const fragment = document.createRange().createContextualFragment(`
+    <div class="product-details__wrapper">
+      <div class="product-details__alert"></div>
+      <div class="product-details__left-column">
+        <div class="product-details__gallery"></div>
+      </div>
+      <div class="product-details__right-column">
+        <div class="product-details__header"></div>
+        <div class="product-details__price"></div>
+        <div class="product-details__gallery"></div>
+        <div class="product-details__short-description"></div>
+        <div class="product-details__configuration">
+          <div class="product-details__options"></div>
+          <div class="product-details__quantity"></div>
+          <div class="product-details__buttons">
+            <div class="product-details__buttons__add-to-cart"></div>
+            <div class="product-details__buttons__add-to-wishlist"></div>
+          </div>
+        </div>
+        <div class="product-details__description"></div>
+        <div class="product-details__attributes"></div>
+      </div>
+    </div>
+  `);
+
+  const $alert = fragment.querySelector('.product-details__alert');
+  const $gallery = fragment.querySelector('.product-details__gallery');
+  const $header = fragment.querySelector('.product-details__header');
+  const $price = fragment.querySelector('.product-details__price');
+  const $galleryMobile = fragment.querySelector('.product-details__right-column .product-details__gallery');
+  const $shortDescription = fragment.querySelector('.product-details__short-description');
+  const $options = fragment.querySelector('.product-details__options');
+  const $quantity = fragment.querySelector('.product-details__quantity');
+  const $addToCart = fragment.querySelector('.product-details__buttons__add-to-cart');
+  const $addToWishlist = fragment.querySelector('.product-details__buttons__add-to-wishlist');
+  const $description = fragment.querySelector('.product-details__description');
+  const $attributes = fragment.querySelector('.product-details__attributes');
+
+  block.appendChild(fragment);
+
+  // Alert
+  let inlineAlert = null;
+
+  // Gallery (Mobile)
+  await PDPProvider.render(ProductGallery, {
```

Comment:
As discussed on the call earlier, each of these `PDPProvider.render` lines is now sync and waiting for the prior one to complete. If there are no internal dependencies between the containers, we may as well wrap all in a resolving Promise like.

```
await Promise.all([
  await PDPProvider.render(...),
  await PDPProvider.render(...),
  ..., 
]);
```

Reply:
Done